src/main/java/com/rabbitmq/client/impl/AMQConnection.java

```diff
@@ -554,7 +559,17 @@ public void run() {
                                     // be discarded.
                                     ChannelManager cm = _channelManager;
                                     if (cm != null) {
-                                        cm.getChannel(frame.channel).handleFrame(frame);
+                                        ChannelN channel;
+                                        try {
+                                            channel = cm.getChannel(frame.channel);
+                                        } catch(UnknownChannelException e) {
+                                            // this can happen if channel has been closed,
+                                            // but there was e.g. an in-flight delivery.
+                                            // just ignoring the frame to avoid closing the whole connection
+                                            LOGGER.info("Received a frame on an unknown channel, ignoring it");
```

Comment:
Sorry, I didn't mean that you should use this message exactly. I think logging channel ID of the frame is still a good idea :)